src/pages/todo/components/TodoListItem.js

```diff
@@ -1,34 +1,89 @@
-import React from 'react';
+import React from 'react'
 import { Link } from 'react-router'
+import { UPDATE_TODO } from '../actions'
 
 class TodoListItem extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {editing: false}
+    this.state = {
+      form: { 
+        title: this.props.todo.title
+      },
+      editing: false
+    }
   }
   
   componentDidUpdate() {
     this.refs.textbox.focus()
   }
+
   _showInput() {
     this.setState({editing: true})
   }
 
-  _hideInput() {
+  _updateTodo(event) {
+    const { todo } = this.props
+    let data
+    switch (event.target.type) {
+      case 'checkbox':
+        data = {
+          title: todo.title,
+          marked: (todo.marked == 1) ? 0 : 1
+        }
+        break;
+      case 'text':
+        data = {title: event.target.value}
+        break;
+    }
+    if(this._validate(this.state.form)) {
+      this.props.dispatch({type: UPDATE_TODO.REQUEST, data: data, id: todo.id})
+    }
     this.setState({editing: false})
   }
```

Comment:
As in TodoList.js, this.setState({editing: false}) should be put in componentWillReceiveProps with async callback.